### DISCUSSION TITLE
Access env variables in frontmatter

### DISCUSSION THREAD

**Discussion by linusang-spark** [vuejs/vitepress#3529] [Q&A]
Would like to know if it is possible to access .env variables inside the frontmatter section of .md files?

**Comment by brc-dd** [comment]
Not supported. You can use transformPageData hook to manually replace things.

**Comment by Karlstens** [comment]
What's your usage case?

I personally use the cross-env npm package to define VITE_TARGET_PROJECT, which allows my VitePress scripts to run explicitly as per that variable. (in my project, a public facing project, and then the same project but privately facing)

"docs-example_a:dev": "cross-env VITE_TARGET_PROJECT=example_a vitepress dev docs",
"docs-example_b:dev": "cross-env VITE_TARGET_PROJECT=example_b vitepress dev docs",
"docs-example_a:build": "cross-env VITE_TARGET_PROJECT=example_a vitepress dev docs",
"docs-example_b:build": "cross-env VITE_TARGET_PROJECT=example_b vitepress dev docs"

I've been wondering how to properly employ node.js .env support for a vitepress project, without the use of the cross-env package. Although not necessarily for use in frontmatter, but rather just in index.js, posts.data.js, config.js and vue components.
